sst/include/SimengCoreWrapper.hh

```diff
@@ -0,0 +1,143 @@
+// clang-format off
+// DO NOT MOVE FROM TOP OF FILE - https://github.com/sstsimulator/sst-core/issues/865
+#include <sst/core/sst_config.h>
+// clang-format on
+#include <sst/core/component.h>
+#include <sst/core/eli/elementinfo.h>
+#include <sst/core/interfaces/stdMem.h>
+
+#include <chrono>
+#include <cmath>
+#include <cstring>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+#include "SimengMemInterface.hh"
+#include "simeng/AlwaysNotTakenPredictor.hh"
+#include "simeng/BTBPredictor.hh"
+#include "simeng/BTB_BWTPredictor.hh"
+#include "simeng/Core.hh"
+#include "simeng/Elf.hh"
+#include "simeng/FixedLatencyMemoryInterface.hh"
+#include "simeng/FlatMemoryInterface.hh"
+#include "simeng/MemoryInterface.hh"
+#include "simeng/ModelConfig.hh"
+#include "simeng/SpecialFileDirGen.hh"
+#include "simeng/arch/Architecture.hh"
+#include "simeng/arch/aarch64/Architecture.hh"
+#include "simeng/arch/aarch64/Instruction.hh"
+#include "simeng/arch/aarch64/MicroDecoder.hh"
+#include "simeng/kernel/Linux.hh"
+#include "simeng/models/emulation/Core.hh"
+#include "simeng/models/inorder/Core.hh"
+#include "simeng/models/outoforder/Core.hh"
+#include "simeng/pipeline/A64FXPortAllocator.hh"
+#include "simeng/pipeline/BalancedPortAllocator.hh"
+#include "simeng/version.hh"
+
+using namespace SST;
+using namespace SST::Interfaces;
+using namespace SST::SSTSimeng;
+using namespace simeng;
+
+namespace SST {
+
+namespace SSTSimeng {
+
+class SimengCoreWrapper : public SST::Component {
```

Comment:
As you did with `sst/include/SimengMemInterface.hh`, needs a comment explaining the class similar to existing header files